tests/test_authentication.py

```diff
@@ -140,13 +139,17 @@ def test_unknown_creds(self):
                               '^access denied: CredentialsLookupError: ')
 
     def test_alternative_credential_lookup(self):
+        # Temporarily use alternate auth
+        old_auth = self.auth
+        self.auth = AlternateAuthentication()
```

Comment:
I don't see any reason for assigning this to `self.auth`. As far as I can see, just assign it to `auth` and change your assertion down below to:
````py
assert isinstance(auth.authenticate(req)[0], ...)
````
Am I missing something?